functions.js

```diff
@@ -1 +1,97 @@
-// YOUR CODE HERE
+// maxOrMin(num1, num2, max)
+// This works but it's pretty smelly:
+
+function maxOrMin(num1, num2, max){
+    if ( max && (num1 > num2) ){
+        return num1;
+    } else if ( max && (num1 < num2) ){
+        return num2;
+    } else if ( max && (num1 > num2) ){
+        return num2;
+    } else {
+        return num1;
+    }
+}
+
+maxOrMin(5,4,false);
+
+//sillySum(arr)
+// Got really stuck on this one, had hasBeenSummed equal to 0 instead of an empty array.
+
+function sillySum(arr){
+  var hasBeenSummed = [];
+  arr.forEach(function sum(elem, index) {
+    hasBeenSummed.push (elem * index);
+  });
+  return (hasBeenSummed);
+}
+
+console.log(sillySum([2,4,6,8]));
+
+// numSquare(max)
+// I'm pretty sure there's a better way of doing this -
+// I'm processing considerably more numbers than I'm using... but after really struggling with this one, this approach worked!!!
+// Kinda lol'ing at it's ridiculousness, though.
+
+function numSquare(max){
+  var hasBeenSquared = [];
+  function stoppingPoint(value){
+      return value <= max;
+  }
+  for (i = 0; i < max; i++){
+    if ( (i * i))
+    hasBeenSquared.push (i * i); // creates an array of every number <= max, squared. In other words, if max is 100, there will be 100 values in the array. Yikes.
+  }
+  return (hasBeenSquared.filter(stoppingPoint)); //chops off every value past max from the array
+}
+
+console.log(numSquare(100));
+
+// isPrime(num)
+
+function isPrime(num){
+  if (num < 2){
+    return false;
+  }else {
+    for (i = 2; i < num; i++) {
+      if (num % i === 0){
+        return false;
+      } else {
+        return true;
+      }
+    }
+  }
+}
+
+// primes(max)
+// Couldn't figure this one out.  Here's what I ended up with.
+
+
```

Comment:
declare variables that are only going to be used by the function inside of the function itself